File: src/main/java/corp/soyummy/auth/AuthRepository.java
package corp.soyummy.auth;

import org.springframework.stereotype.Repository;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

@Repository
public interface AuthRepository extends MongoRepository<User, String> {
    Optional<User> findByEmail(String email);
    Optional<User> findByToken(String token);
}

File: src/main/kotlin/com/soyummy/auth/AuthRepository.kt
package com.soyummy.auth

import org.springframework.stereotype.Repository
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.Optional

@Repository
interface AuthRepository : MongoRepository<User, String> {
    fun findByEmail(email: String): Optional<User>
    fun findByToken(token: String): Optional<User>
}